modules/apps/27-interchain-accounts/controller/keeper/migrations.go

```diff
@@ -55,8 +55,7 @@ func (m Migrator) AssertChannelCapabilityMigrations(ctx sdk.Context) error {
 // MigrateParams migrates the controller submodule's parameters from the x/params to self store.
 func (m Migrator) MigrateParams(ctx sdk.Context) error {
 	if m.keeper != nil {
-		var params controllertypes.Params
-		m.keeper.legacySubspace.GetParamSet(ctx, &params)
+		params := m.keeper.GetParams(ctx)
```

Comment:
I believe this migration can no longer be successfully performed, so I think we should panic if a chain tries to execute it. (I would remove all the code within the `if m.keeper != nil` block and replace it with a panic, indicating that a in-place migration must occur to the `v8.x` line before upgrading to `v9.x`

Reply:
@colin-axner Thanks for pointing that for me but I think we should return `ErrInvalidVersion` instead panic because this will not change function return type then we don't need to remove migration test and after this function, it still panic if `MigrateParams` be executed: